api/src/certification/enrolment/infrastructure/repositories/certification-candidate-repository.js

```diff
@@ -153,21 +153,36 @@ const getWithComplementaryCertification = async function (id) {
   return _toDomain(candidateData);
 };
 
-const findCertificationCandidateCompanionInfoByUserId = async function ({ userId }) {
-  const result = await knex.select('sessionId', 'id').from('certification-candidates').where({ userId }).first();
-  if (!result) {
-    throw new NotFoundError(`User ${userId} is not found in a certification's session`);
+const findCompanionPingInfoByUserId = async function ({ userId }) {
+  const knexConn = DomainTransaction.getConnection();
+
+  const latestCertificationCourse = await knexConn
+    .select('sessionId', 'completedAt', 'endedAt')
+    .from('certification-courses')
+    .where({ userId })
+    .orderBy('createdAt', 'desc')
+    .first();
+
+  if (latestCertificationCourse && !latestCertificationCourse.completedAt && !latestCertificationCourse.endedAt) {
+    const { sessionId } = latestCertificationCourse;
+    const { id: certificationCandidateId } = await knex
```

Comment:
Utilisation la même instance de `knex` pour récupérer la transaction

```suggestion
const { id: certificationCandidateId } = await knexConn
```